applications/serial_lte_modem/src/main.c

```diff
@@ -54,6 +54,35 @@ void nrf_modem_recoverable_error_handler(uint32_t err)
 	LOG_ERR("Modem library recoverable error: %u", err);
 }
 
+static int ext_xtal_control(bool on_off)
+{
+#if defined(CONFIG_SLM_EXTERNAL_XTAL)
+	int err;
+	static struct onoff_client cli = {};
```

Comment:
since you are busywaiting it does not need to be static. Same with `clk_mgr` as you get it from clock control on every call.

Reply:
Updated, thanks.

Reply:
@nordic-krch please comment whether current implementation acceptable or not?

Reply:
yes, i missed that you are not busywaiting for clock to be on, just initialize cli and leave the function. In that case, it must be static as for manager variable, if you would use `z_nrf_clock_control_get_onoff` in release then it could be turned to automatic variable but it can stay as it is now.

One thing, since you are not waiting for clock to be ready, it may be possible that attempt to release the clock will happen before clock is turned on. In that case `onoff_release` will fail because it assumes that service is already on. You need to use `onoff_cancel_or_release(clk_mgr, &cli)` which takes care of both cases (cancels request or releases).

Reply:
Thanks for the details, in our case the release happens after power off UART, which means the clock must have been turned on before (or UART won't work). Anyway, let me test ``onoff_cancel_or_release()`` then come back with result.